Loop/Managers/DeviceDataManager.swift

```diff
@@ -1308,3 +1322,20 @@ extension DeviceDataManager: SupportInfoProvider {
     }
     
 }
+
+extension DeviceDataManager: GlucoseUnitPublisher {
+    func addGlucoseUnitObserver(_ observer: GlucoseUnitObserver, queue: DispatchQueue = .main) {
+        glucoseUnitObservers.insert(observer, queue: queue)
+        if let glucoseUnit = glucoseStore.preferredUnit {
+            observer.glucoseUnitDidChange(to: glucoseUnit)
```

Comment:
Does this need to be dispatched on the `queue` specified (if we aren't already on that queue)?